background/services/chain/index.ts

```diff
@@ -346,6 +321,89 @@ export default class ChainService extends BaseService<Events> {
       : this.providers.evm[network.chainID]
   }
 
+  /**
+   * Pulls the list of active networks from memory or indexedDB.
+   * Defaults to ethereum in the case that neither exist.
+   */
+  async getActiveNetworks(): Promise<EVMNetwork[]> {
+    if (this.activeNetworks.length > 0) {
+      return this.activeNetworks
+    }
+
+    // Since activeNetworks will be an empty array at extension load (or reload time)
+    // we need a durable way to track which networks an extension is tracking.
+    // The below code should only be called once per extension reload for extensions
+    // with active accounts
+    const networksToTrack = await this.getNetworksToTrack()
+    if (networksToTrack.length > 0) {
+      networksToTrack.forEach((network) => {
+        this.activateNetworkOrThrow(network.chainID)
+      })
+      return this.activeNetworks
+    }
+
+    // Default to supporting Ethereum so ENS resolution works during onboarding
+    this.activateNetworkOrThrow(ETHEREUM.chainID)
+    return this.activeNetworks
+  }
+
+  async subscribeToNetworkEvents(network: EVMNetwork): Promise<void> {
+    const provider = this.providerForNetwork(network)
+    if (provider) {
+      await Promise.allSettled([
+        this.fetchLatestBlockForNetwork(network),
+        this.subscribeToNewHeads(network),
+      ])
+    } else {
+      logger.error(`Couldn't find provider for network ${network.name}`)
+    }
+  }
+
+  /**
+   * Adds a supported network to list of active networks.
+   */
+  async activateNetworkOrThrow(chainID: string): Promise<EVMNetwork> {
+    const activeNetwork = this.activeNetworks.find(
+      (ntwrk) => toHexChainID(ntwrk.chainID) === toHexChainID(chainID)
+    )
+
+    if (activeNetwork) {
+      logger.warn(
+        `${activeNetwork.name} already active - no need to activate it`
+      )
+      return activeNetwork
+    }
+
+    const networkToActivate = this.supportedNetworks.find(
+      (ntwrk) => toHexChainID(ntwrk.chainID) === toHexChainID(chainID)
+    )
+    if (!networkToActivate) {
+      throw new Error(`Network with chainID ${chainID} is not supported`)
+    }
+
+    this.activeNetworks.push(networkToActivate)
+
+    const existingSubscription = this.subscribedNetworks.find(
+      (networkSubscription) =>
+        networkSubscription.network.chainID === networkToActivate.chainID
+    )
+
+    if (!existingSubscription) {
+      this.subscribeToNetworkEvents(networkToActivate)
+      const addressesToTrack = new Set([
+        ...(await this.getAccountsToTrack()).map((account) => account.address),
+      ])
```

Comment:
```suggestion
      const addressesToTrack = new Set(
        await this.getAccountsToTrack()).map((account) => account.address),
      )
```

I expect this would work without the extra array unwrap/rewrap.